### DISCUSSION TITLE
Show loader while lazy component's code is in fetching mode

### DISCUSSION THREAD

**Discussion by scofield-ua** [nuxt/nuxt#20782] [Questions]
Hey, is there is any way (hooks or something) to detect when [lazy](https://nuxt.com/docs/guide/directory-structure/components#dynamic-imports) component's code is loading? 

So I can show some "loading indication" while the code is still loading.

And is it possible to detect it globally not per component?

**Comment by nWacky** [comment]
You can use [defineAsyncComponent](https://vuejs.org/api/general.html#defineasynccomponent)

```vue
<script setup>
import { ref, defineComponent, h, defineAsyncComponent } from 'vue';
import { ComponentA } from "#components";

const AsyncComponent = defineAsyncComponent({
  loader: () => new Promise((res) => setTimeout(res, 5000)).then(ComponentB),
  loadingComponent: ComponentA,
  delay: 2000,
  suspensible: false,
});
</script>

<template>
  <ClientOnly>
    <AsyncComponent />
  </ClientOnly>
</template>
```

[original comment by LinusBorg](https://github.com/nuxt/nuxt/issues/15648#issuecomment-1397434396)
